src/main/java/be/seeseemelk/mockbukkit/entity/BoatMock.java

```
@@ -0,0 +1,171 @@
+	private double unoccupiedDeceleration = -1;
+	private boolean workOnLand = false;
+
+	/**
+	 * Constructs a new {@link BoatMock} on the provided {@link ServerMock} with a specified {@link UUID}.
+	 *
+	 * @param server The server to create the entity on.
+	 * @param uuid   The UUID of the entity.
+	 */
+	public BoatMock(@NotNull ServerMock server, @NotNull UUID uuid)
+	{
+		super(server, uuid);
+	}
+
+	@Override
+	@Deprecated
+	public @NotNull TreeSpecies getWoodType()
+	{
+		return switch (boatType)
+		{
+			case SPRUCE -> TreeSpecies.REDWOOD;
+			case BIRCH -> TreeSpecies.BIRCH;
+			case JUNGLE -> TreeSpecies.JUNGLE;
+			case ACACIA -> TreeSpecies.ACACIA;
+			case DARK_OAK -> TreeSpecies.DARK_OAK;
+			default -> TreeSpecies.GENERIC;
+		};
+	}
+
+	@Override
+	@Deprecated
+	public void setWoodType(@NotNull TreeSpecies species)
+	{
+		Preconditions.checkNotNull(species, "Species can not be null");
+		Type type;
+
+		type = switch (species)
+		{
+			case GENERIC -> Type.OAK;
+			case BIRCH -> Type.BIRCH;
+			case ACACIA -> Type.ACACIA;
+			case JUNGLE -> Type.JUNGLE;
+			case DARK_OAK -> Type.DARK_OAK;
+			case REDWOOD -> Type.SPRUCE;
+		};
+
+		this.boatType = type;
+
+	}
+
+	@Override
+	public @NotNull Type getBoatType()
+	{
+		return this.boatType;
+	}
+
+	@Override
+	public void setBoatType(@NotNull Type type)
+	{
+		Preconditions.checkNotNull(type, "Type cannot be null");
+		this.boatType = type;
+	}
+
+	@Override
+	@Deprecated
+	public double getMaxSpeed()
+	{
+		return this.maxSpeed;
+	}
+
+	@Override
+	@Deprecated
+	public void setMaxSpeed(double speed)
+	{
+		Preconditions.checkArgument(speed >= 0.0D, "Speed cannot be negative");
+		this.maxSpeed = speed;
+	}
+
+	@Override
+	@Deprecated
+	public double getOccupiedDeceleration()
+	{
+		return this.ocupiedDeceleration;
+	}
+
+	@Override
+	@Deprecated
+	public void setOccupiedDeceleration(double rate)
+	{
+		Preconditions.checkArgument(rate >= 0.0D, "Rate cannot be negative");
```

Same here, it silently fails in paper